### ISSUE TITLE
Confusing exception when model would contain a relationship between entities with only shadow keys

### ISSUE THREAD

**Issue by divega** [dotnet/efcore#5655]
### Steps to reproduce

``` C#
using System;
using System.Collections.Generic;
using Microsoft.EntityFrameworkCore;
using Xunit;

namespace TestShadowKey
{
    public class Tests
    {
        [Fact]
        public void Relationship_with_automatic_shadow_keys_throws_confusing_exception()
        {
            using (var context = new MyContext())
            {
                Assert.EndsWith(
                    @"Configure a non-shadow principal key for this relationship.",
                    Assert.Throws<InvalidOperationException>(() => context.Model).Message);
            }
        }

        public class MyContext : DbContext
        {
            public DbSet<Customer> Customers { get; set; }
            public DbSet<Order> Orders { get; set; }

            protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
            {
                optionsBuilder.UseSqlServer(
                    "Data Source=(localdb)\\mssqllocaldb;Initial Catalog=blah;Integrated Security=true");
            }

        public class Customer
        {
            public IList<Order> Orders { get; } = new List<Order>();
            public string Name { get; set; }
        }

        public class Order
        {
            public Customer Customer { get; set; }
            public string Description { get; set; }
        }
    }
}
```

Note this is similar to an existing test that uses shadow entities: [`Can_create_shadow_navigations_between_shadow_entity_types`](https://github.com/aspnet/EntityFramework/blob/e72c4a178f6d2706ac6c65807dcb7a86042de96c/test/Microsoft.EntityFrameworkCore.Tests/ModelBuilderTest/ModelBuilderNonGenericStringTest.cs#L21)
### The issue

This is a negative scenario because we require that shadow primary keys are explicitly configured, but the exception message is still confusing:

> The key {'TempId'} contains properties in shadow state and is referenced by a relationship from 'Customer.Orders' to 'Order.Customer'. Configure a non-shadow principal key for this relationship.

a. `{'TempId'}` is not in user code, and it is just an obscure implementation detail of our conventions.  
b. The suggestion to create non-shadow keys is not the only solution. Explicitly creating the shadow keys also works:

``` C#
            protected override void OnModelCreating(ModelBuilder modelBuilder)
            {
                modelBuilder.Entity<Customer>(builder =>
                {
                    builder.Property<int>("Id");
                    builder.HasKey("Id");
                });
                modelBuilder.Entity<Order>(builder =>
                {
                    builder.Property<int>("Id");
                    builder.HasKey("Id");
                });
            }
```
### Further technical details

EF Core version: 1.0.0-rc2-final or newer
Operating system: Windows
Visual Studio version: 2015 Update 2

**Comment by divega** [comment]
cc @danroth27

**Comment by KelseySteele** [comment]
I'm creating four unrelated tables . I get the dreaded shadow property error as shown below. 

System.InvalidOperationException: The key {'TempId'} contains properties in shadow state and is referenced by a relationship from 'HttpContext' to 'ControllerContext.HttpContext'. Configure a non-shadow principal key for this relationship.    at Microsoft.EntityFrameworkCore.Metadata.Conventions.Internal.KeyConvention.Apply(InternalModelBuilder modelBuilder)
   at Microsoft.EntityFrameworkCore.Metadata.Conventions.Internal.ConventionDispatcher.OnModelBuilt(InternalModelBuilder modelBuilder)
   at Microsoft.EntityFrameworkCore.Infrastructure.ModelSource.CreateModel(DbContext context, IConventionSetBuilder conventionSetBuilder, IModelValidator validator)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
    at Microsoft.EntityFrameworkCore.Internal.DbContextServices.CreateModel()
   at Microsoft.EntityFrameworkCore.Internal.LazyRef`1.get_Value()
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.ScopedCallSite.Invoke(ServiceProvider provider)
    at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetService[T](IServiceProvider provider) 
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.TransientCallSite.Invoke(ServiceProvider provider)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ConstructorCallSite.Invoke(ServiceProvider provider)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.TransientCallSite.Invoke(ServiceProvider provider)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService(IServiceProvider provider, Type serviceType)
    at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService[T](IServiceProvider provider)
   at Microsoft.EntityFrameworkCore.Design.MigrationsOperations.AddMigration(String name, String outputDir, String contextType)
    at Microsoft.EntityFrameworkCore.Tools.Cli.MigrationsAddCommand.Execute(CommonOptions commonOptions, String name, String outputDir, String context, String environment, Action`1 reporter)
   at Microsoft.EntityFrameworkCore.Tools.Cli.MigrationsAddCommand.<>c__DisplayClass0_0.<Configure>b__0()
    at Microsoft.Extensions.CommandLineUtils.CommandLineApplication.Execute(String[] args)
   at Microsoft.EntityFrameworkCore.Tools.Cli.Program.Main(String[] args)
 The key {'TempId'} contains properties in shadow state and is referenced by a relationship from 'HttpContext' to 'ControllerContext.HttpContext'. Configure a non-shadow principal key for this relationship. 

Is there a way to correct this? Do you need more information?

**Comment by divega** [comment]
@KelseySteele If your model doesn't match the scenarios in #5655 or #6260 it could be a different bug. In that case please a new issue with the needed details to repro it.

**Comment by divega** [comment]
Closing this since we are planning to improve the exception message as part of the fix for #6260 in 1.0.1. 

@rowanmiller at some point before we ship 1.1.0 we should do a wholesale review of string changes.
